src/Raven.Server/Documents/Queries/Dynamic/CollectionQueryRunner.cs

```diff
@@ -203,19 +204,20 @@ private unsafe void FillCountOfResultsAndIndexEtag(QueryResultServerSide<Documen
                 var numberOfDocuments = Database.DocumentsStorage.GetNumberOfDocuments(context);
                 buffer[0] = DocumentsStorage.ReadLastDocumentEtag(context.Transaction.InnerTransaction);
                 buffer[1] = DocumentsStorage.ReadLastTombstoneEtag(context.Transaction.InnerTransaction);
-                buffer[2] = numberOfDocuments;
+                buffer[2] = DocumentsStorage.ReadLastCountersEtag(context.Transaction.InnerTransaction);
```

Comment:
This needs to only be applied when we have include on counters inside the query. We need to detect that and put that information inside the QueryMetadata. If we will not do that then counters will impact query caching mechanism pretty badly.